spec/src/main/asciidoc/configprofile.asciidoc

```diff
@@ -0,0 +1,80 @@
+This is because the name of the profile is directly stored in the name of the config property. e.g.
+
+[source, text]
+----
+java -jar myapp.jar -Dmp.config.profile=testing
+----
+
+The property `mp.config.profile` is only read once on microservice starting. Any changes after that is ignored. Furthermore, the value specifies a single profile. Only one profile can be active at a time.
+Commas in the value have no special meaning. For example, if the value of `mp.config.profile` is `testing,live`,  a single profile named `testing,live` is enabled instead of enabling two profiles. 
+
+=== How Config Profile works
+
+In supporting the Config Profile, the config property names will need to start with the naming convention of `%configProfile.`, (e.g. `%live.vehicle.name`).
+
+For an instance, a config source can be specified as follows.
+
+[source, text]
+----
+%dev.vehicle.name=car
+%live.vehicle.name=train
+%testing.vehicle.name=bike
+vehicle.name=lorry
+----
+
+A config property associated with the Config Profile can be looked up as follows.
+
+[source, text]
+----
+@Inject @ConfigProperty(name="vehicle.name") String vehicleName;
+----
+
+[source, text]
+----
+String vehicleName = ConfigProvider.getConfig().getValue("vehicle.name", String.class);
+----
+
+If the system property `mp.config.profile` is set to `dev`, the property `%dev.vehicle.name` is the Active Property. An active property overrides the properties in the same config source. 
+In more details, if the system property `mp.config.profile` is set to `dev`, the property `%dev.vehicle.name` overrides the property `vehicle.name`. 
+The properties `%live.vehicle.name` and `%testing.vehicle.name` are inactive config property and don't override the property `vehicle.name`.
+
+In more details, on setting the system property `mp.config.profile` is set to `dev`, the property `%dev.vehicle.name` is the active property. The `vehicleName` is `train`.
+On setting the system property `mp.config.profile` is set to `live`, the property `%live.vehicle.name` is the active property. The `vehicleName` is `train`. Similarly, `bike` will be the value of `vehicleName`, if the profile is `testing`.
+
+When retrieving the config values, this Config spec implementations must first search for the value of `mp.config.profile` and then retrieve the value of the config property `%${mp.config.profile}.vehicle.name`,
+with the consideration of config source hierarchy specified via ordinal. It means if more than one config source contains the same property `%${mp.config.profile}.vehicle.name`, 
+the value from the config source with the highest ordinal will be used. In the same config source, the property `%${mp.config.profile}.vehicle.name` overrides the config property `vehicle.name`. 
+If no value is found, the value of the config name `vehicle.name` will be retrieved.
```

Comment:
I think all this is too complicated. I suggest to describe the algorithm as follows:

> Conforming implementations are required to search for a configuration source with the highest ordinal (priority) that provides either the property name or the "profile-specific" property name. If the configuration source provides the "profile-specific" name, the value of the "profile-specific" property will be returned. If it doesn't contain the "profile-specific" name, the value of the plain property will be returned. A "profile-specific" property name is defined as a property whose name consists of the following sequence: `% <profile name> . <original property name>`

Reply:
So does the search stop if the highest ordinal config source returns a non-profile specific value for the property?

Reply:
Yes, that's the idea. We discussed this in this thread: https://github.com/eclipse/microprofile-config/pull/554/files#r409673972. And agreed that "profile properties override base properties only in the same config source.". So the search can stop with the highest ordinal config source that provides the property, even if it's a non-profile specific property.

This is more efficient to implement. It's also less confusing for users because it's easier to find a profile-specific property that overrides the default value because it has to be in the same config source.

Reply:
I rephrased as @OndroMih suggested.